spec/infinispan_local_spec.js

```diff
@@ -5,370 +5,274 @@ var readFile = Promise.denodeify(require('fs').readFile);
 var f = require('../lib/functional');
 var t = require('./utils/testing'); // Testing dependency
 
-describe('Infinispan local client', function() {
-  var client = t.client(t.local);
+function describeInfinispanBehaviourInLocalModeWithDifferentHotrodProtocol(options) {
+  describe('with hotrod protocol: ' + options.version, function () {
+    var client = t.client(t.local, null, options.version);
 
-  beforeEach(function(done) { client
```

Comment:
+1 to have an environment variable to drive testing particular protocol version.

> I have played a bit with this argument, and seems that after running the testsuite with e.g. --config A B, the process.env.A = B . Then the testsuite is run without specifying the config, still the process.env.A = B. I couldn't find a way of unsetting it though, if only passing another value for A while running the test suite or deleting it manually after the testsuite is finished (delete process.env.A;) -> not sure how to catch this event.

Ok, if we can't get around that limitation, we can always have that as a CI or test configuration. So when you test it, you'd execute as a two step process:

    $ jasmine-node spec // <- should run with protocol 2.5
    $ jasmine-node spec --config protocol 2.2

The only thing to remember is that some tests won't work with older versions, so we'd have to exclude those somehow.